pallets/file-system/src/utils.rs

```diff
@@ -551,6 +559,7 @@ where
             // Issue a challenge to force the BSPs to update their storage root.
             <T::ProofDealer as shp_traits::ProofsDealerInterface>::challenge_with_priority(
                 &file_key,
+                ChallengeKeyInclusion::NotIncluded,
```

Comment:
Since we are removing a file key (deleting a storage request), we could force providers to generate a proof of non-inclusion for the given file key. This would force all providers to delete the file key from their forest to successfully respond to the challenge. 

This would not give the runtime the possibility of applying the delta to the Forest itself and updating the root. What we can do potentially if we want to be consistent (having the runtime apply the delta) is to specify add an extra flag to the challenge that states whether or not the file should be deleted or added (basically adding a mutation as well to the challenge). This would mean we specify `None` instead of `ChallengeKeyInclusion::NotIncluded` and whenever a provider submits a proof that includes the file key, the runtime will apply the delta and update the root.

The challenges (including priority ones) would have a storage type that resembles something like this:

```rust
type Challenges = (Key, Option<ChallengeKeyInclusion>, Delta)
```